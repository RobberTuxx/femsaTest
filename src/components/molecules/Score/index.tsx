import Score from "./view";

export default Score
