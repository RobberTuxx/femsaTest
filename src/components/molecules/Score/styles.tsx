import {StyleSheet} from "react-native";
import {Colors} from "../../../styles/theme";
import {heightResponsive, widthResponsive} from "../../../utils/responsive";

export const styles = StyleSheet.create({
    container: {
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: Colors.blue,
        width: widthResponsive(286),
        height: heightResponsive(143),
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,

        elevation: 5,
    },
    points: {
        fontSize: heightResponsive(32),
        fontWeight: "900",
        fontFamily: "Avenir",
        color: Colors.white
    },
    month: {
        fontSize: heightResponsive(16),
        fontWeight: "900",
        fontFamily: "Avenir",
        color: Colors.white,
        marginLeft: widthResponsive(-180),
        marginTop: heightResponsive(-20)
    }
})
