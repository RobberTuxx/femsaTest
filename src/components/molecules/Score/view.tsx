import React from "react";
import {Text, View} from "react-native";
import {styles} from "./styles";

interface ScoreProps {
    points: number
}

export default function Score(props: ScoreProps) {
    return (
        <View style={styles.container}>
            <Text style={styles.month}>Diciembre</Text>
            <Text style={styles.points}>{props.points?.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')} pts</Text>
        </View>
    )
}
