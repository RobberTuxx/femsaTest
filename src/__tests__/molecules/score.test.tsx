import React from 'react';
import renderer from 'react-test-renderer';
import Score from "../../components/molecules/Score";
it('render score', ()=> {
    const component = renderer.create(<Score points={10000}/>).toJSON()
    expect(component).toMatchSnapshot()
})
it('has 0 child', ()=> {
    const component = renderer.create(<Score points={10000}/>).toJSON()
    expect(component.children.length).toBe(2)
})
