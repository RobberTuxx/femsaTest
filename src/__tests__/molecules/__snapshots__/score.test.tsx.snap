// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 20,
      "elevation": 5,
      "height": 224,
      "justifyContent": "center",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 546,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Avenir",
        "fontSize": 25,
        "fontWeight": "900",
        "marginLeft": -343.5,
        "marginTop": -31.5,
      }
    }
  >
    Diciembre
  </Text>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Avenir",
        "fontSize": 50,
        "fontWeight": "900",
      }
    }
  >
    10,000.00
     pts
  </Text>
</View>
`;

exports[`render score 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 20,
      "elevation": 5,
      "height": 224,
      "justifyContent": "center",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 546,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Avenir",
        "fontSize": 25,
        "fontWeight": "900",
        "marginLeft": -343.5,
        "marginTop": -31.5,
      }
    }
  >
    Diciembre
  </Text>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Avenir",
        "fontSize": 50,
        "fontWeight": "900",
      }
    }
  >
    10,000.00
     pts
  </Text>
</View>
`;
